From 0db8dc6022f67a4e1f49397b8bf519b2a34f74c9 Mon Sep 17 00:00:00 2001
From: Jelle van der Waa <jelle@vdwaa.nl>
Date: Wed, 27 Dec 2017 11:05:45 +0100
Subject: [PATCH] Replace removed is_error macro

The json-c library removed the is_error macro in 0.13, replaced the
macro calls with a != NULL statement.
---
 bti.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/bti.c b/bti.c
index 7f485a8..9b68e17 100644
--- a/bti.c
+++ b/bti.c
@@ -451,7 +451,7 @@ static void parse_timeline(char *document, struct session *session)
 	struct json_object *val;				\
 	struct lh_entry *entry;					\
 	for (entry = json_object_get_object(obj)->head;		\
-		({ if(entry && !is_error(entry)) {		\
+		({ if(entry && entry != NULL) {		\
 			key = (char*)entry->k;			\
 			val = (struct json_object*)entry->v;	\
 		} ; entry; });					\
@@ -667,7 +667,7 @@ static int parse_response_json(char *document, struct session *session)
 
 	/* make global for now */
 	store_session = session;
-	if (!is_error(jobj)) {
+	if (jobj != NULL) {
 		/* guards against a json pre 0.10 bug */
 		json_parse(jobj,0);
 	}
@@ -692,7 +692,7 @@ static void parse_timeline_json(char *document, struct session *session)
 
 	/* make global for now */
 	store_session = session;
-	if (!is_error(jobj)) {
+	if (jobj != NULL) {
 		/* guards against a json pre 0.10 bug */
 		if (json_object_get_type(jobj)==json_type_array) {
 			json_parse_array(jobj, NULL, 0);
